### PR TITLE
Expose surface inputs decal

### PR DESCRIPTION
**Pull request by alelievr**
### Purpose of this PR
Exposed the surface inputs of decals from ShaderGraph to the material.
Decal

Note that if the affects* option is ticked in the ShaderGraph, then the keyword to switch it on/off on the material will be generated. 

Decal shader diff:
https://www.diffchecker.com/0Yzd2ZKr

---
### Testing status
HDRP decal tests are green locally.

### CODE REVIEW

**Comment by sebastienlagarde** [comment]
@alelievr the Emissive shader feature shold only be use for the projector and decal mesh emissive, and not
so in the pass projector emissive/ decal mesh we must not have other keyword than emissive.

**Comment by sebastienlagarde** [comment]
The stencil mask of decal are hard coded.
It should be WriteMask[_DecalStencilWriteMask] intead of WriteMask [16] like in DEcal.shader. 
Note: I know it is not part of this PR, just added here as a note to something we need to do

**Comment by sebastienlagarde** [comment]
weird enough, the Z-test is GRrater instead of being ZEqual.  like in Decal.shader.
Weird that it work :), again not due to your PR

**Comment by sebastienlagarde** [comment]
there is a decal project mask for projectr 4RT but not for projector 3 RT, not expected, both can deal with different color mask

File: com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.cs [hunk 1]
```diff
@@ -448,11 +457,11 @@ static class DecalRenderStates
                 { RenderState.Blend("Blend 0 SrcAlpha OneMinusSrcAlpha, Zero OneMinusSrcAlpha Blend 1 SrcAlpha OneMinusSrcAlpha, Zero OneMinusSrcAlpha Blend 2 SrcAlpha OneMinusSrcAlpha, Zero OneMinusSrcAlpha") },
                 { RenderState.ZTest(ZTest.LEqual) },
                 { RenderState.ZWrite(ZWrite.Off) },
-                { RenderState.ColorMask(s_DecalColorMasks[4]) },
+                { RenderState.ColorMask("ColorMask BA 2") },
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.cs:460]: is it expected?

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.cs:460]: harcoded mask value here

File: com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.cs [hunk 2]
```diff
@@ -28,7 +28,7 @@ sealed partial class DecalSubTarget : HDSubTarget, ILegacyTarget, IRequiresData<
         protected override string customInspector => "Rendering.HighDefinition.DecalGUI";
         protected override string renderType => HDRenderTypeTags.Opaque.ToString();
         protected override string renderQueue => HDRenderQueue.GetShaderTagValue(HDRenderQueue.ChangeType(HDRenderQueue.RenderQueueType.Opaque, decalData.drawOrder, false, false));
-        protected override ShaderID shaderID => HDShaderUtils.ShaderID.SG_Lit;
+        protected override ShaderID shaderID => HDShaderUtils.ShaderID.SG_Decal;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.cs:31]: Arf. Guess it mean that already converted material will not work, as we can't detect they are decal?

Mean for the migration step we should maybe not test against HDShaderUtils.ShaderID.SG_Decal; but just migrate every material ?

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.cs:31]: or is this property not serialize and just a on the fly thing?

File: com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.cs (hunk 1 above)
Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.cs:460]: There is the same hardcoded value in Decal.shader If I'm not mistaken

File: com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.cs [hunk 3]
```diff
@@ -358,29 +409,19 @@ static class DecalRequiredFields
...
+            readonly static string s_DecalColorMask = "ColorMask [_DecalColorMask2] 2 ColorMask [_DecalColorMask3] 3";
 
             public static RenderStateCollection Projector3RT = new RenderStateCollection
             {
                 { RenderState.Blend("Blend 0 SrcAlpha OneMinusSrcAlpha, Zero OneMinusSrcAlpha Blend 1 SrcAlpha OneMinusSrcAlpha, Zero OneMinusSrcAlpha Blend 2 SrcAlpha OneMinusSrcAlpha, Zero OneMinusSrcAlpha") },
                 { RenderState.Cull(Cull.Front) },
                 { RenderState.ZTest(ZTest.Greater) },
                 { RenderState.ZWrite(ZWrite.Off) },
-                { RenderState.ColorMask(s_DecalColorMasks[4]) },
+                { RenderState.ColorMask("ColorMask BA 2") },
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.cs:420]: Same remark, don't think it is expected

File: com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.cs (hunk 2 above)
Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubTarget.cs:31]: It's not serialized, so it's fine. We never tested the shader id in our upgrade script anyway

**Comment by sebastienlagarde** [comment]
replace by https://github.com/Unity-Technologies/Graphics/pull/1398

**Pull request closed without merging**